### ISSUE TITLE
OpenSSL changes PHP 5.6, options to configure client stream_context

### ISSUE THREAD

**Issue by snapnie** [wsdl2phpgenerator/wsdl2phpgenerator#261]
Hi,

We recently upgraded to php 5.6, due to the OpenSSL changes in this version i was wondering if wsdl2php has a possibility to configure the soapclient's stream_context.

Im currently using wsdl2php version 3.3.1.
I cant find an implementation in StreamContextFactory->create, perhaps im missing something.

Kind regards,
Aldo

**Comment by red-led** [comment]
You can add [soapClientClass](https://github.com/wsdl2phpgenerator/wsdl2phpgenerator#soapclientclass) config option, and use your own, extended from PHP's ```SoapClient``` which will add required stream context.

**Comment by snapnie** [comment]
Ah, nice. Thanks red-led

**Comment by snapnie** [comment]
I think my problem description wasn't really clear, sry about that. let me try again.

Wsdl2phpgenerator expects an inputfile (the wsdl from which php classes are generated). 
The wsdl im using as inputfile has a certificate. Php 5.6 tries to validate this certificate. Apparently  the root ca isnt in the trusted list. So i need to add this ca (and the complete chain) manually.
This can be done by creating a stream context: 
stream_context_create(["ssl"] => ["cafile" => $path_to_cafile]) where $path_to_cafiel is a .pem file with a bundle of ca root certificates.

Currently wsdl2phpgenerator constructs a WsdlDocument (SchemaDocument) and creates a stream context using the StreamContextFactory->create.
As far as i can see there is no option to configure the stream context.

An alternatieve for my situation could be to update the repositories of our serversto update the trusted list with root certificates. But this a lot of work and takes a lot of time (puppetized, different department, blabla ). 

The solution red-led provided solves the problem when consuming the wsdl from the generated classes.

**Comment by kasperg** [comment]
I see no way to achieve this with wsdl2phpgenerator currently.

If you are interesting in contributiing you could expand the Configuration to also access a "raw" array of stream context options. This could then be retrieved in the [`StreamContextFactory`](https://github.com/wsdl2phpgenerator/wsdl2phpgenerator/blob/master/src/StreamContextFactory.php) and used when creating new stream contexts.

**Comment by drm** [comment]
That's what I would suggest :) I ran into this issue as well, I'll make a pull request for it. Do you have a preference on which version to base that?

**Comment by kasperg** [comment]
Sounds good @drm.

You should branch out from master. No other pull request should be affected by this.

Please remember to update the README as well.

**Comment by snapnie** [comment]
@drm, thanks, nice work.
@kasperg: could you release drm's work so that composer can manage the dependencies. Im getting version 3.3.1 (https://packagist.org/packages/wsdl2phpgenerator/wsdl2phpgenerator). Thanks in advance

**Comment by kasperg** [comment]
@snapnie: We are discussing the suggested changes in #262. When the final issues are resolved it will be merged and part of a new release.
